### DISCUSSION TITLE
The (Distant) Future of the Map; Beyond the Galaxy

### DISCUSSION THREAD

**Discussion by AvianGeneticist** [endless-sky/endless-sky#7214] [Ideas]
So there is a **lot** of discussion of the future of how to change or update the map, especially as we find more and more people (myself included) working on race PRs, and others suggesting some other map changes such as adding more uninhabited regions, we begin to run into a bit of an interesting question:
_Where the hell do we put all this?_

Some people (notably Azure) have tried to make adjustments to the galaxy map to increase the real-estate available. Though the efforts are definitely appreciated, I think I'd rather see the Milky Way eventually be "finished". What happens after that? Well, we have easily modifiable galaxies, let's use it!

I'd like to propose that, once the space in the Milky Way is fully occupied (at least to a reasonable extent). The Laniakea supercluster (the cluster the Milky Way belongs to) houses **thousands** of separate galaxies; why not use them? Andromeda is the biggest example, obviously, but there are many dwarf galaxies or star clusters that could be utilized. In this way, we avoid any major issues with the existing galaxy map (such as having to move anything in the Milky Way), while also giving the game a grander scope.

Of course, there are some issues, especially in regards with the act of world traversal. Though I think this will simply become a problem in general as the game gets larger and larger. Wormholes would be a necessity, obviously, except for the **very** rare bridge of star clusters between galaxies (quite unlikely, but not impossible, given the sheer amount of space within the supercluster).

This is all very far away, but I don't think there's any harm in discussing it all ahead of time. If nothing else, I hope this discussion leads to some more alternative ideas to just "rework the map".

**Comment by Azure3141** [comment]
I don't think adding a ton of galaxies would really help, let alone adding the Laniakea Supercluster (seriously, that's over 100k galaxies).

I would prefer it if ES became a deeper game, working with the single galaxy (and its satellites) that we have, rather than stretching it across such a vast expanse that it becomes thin or broadly populated with samey feeling civilizations and stories.

**Reply by RisingLeaf** [reply]
That is a good point but still I would like to see at least one more galaxy, Andromeda, in game. Andromeda is a galaxy that we see at our night sky and I think that makes it worth at least a mention.

**Comment by Zitchas** [comment]
Well, before we get to adding the rest of the cluster, I'd like to see the map density increased a bit. It's always kind of bugged me that each hyperlink crosses something in the order of several thousand light years per jump. Being able to go from one end of the galaxy to the other in roughly 20 days (not counting wormholes) just seems way too fast/small to me. 

Ideally, I'd like that dropped to about the point where each hyperlink is covering about 500-1000 light years; giving us about 40-50 jumps to cross from one edge to the other. That's more than enough space to get another dozen or so races in here at the very least, probably closer to twenty if we hold to the idea that at least some of them should be small 1-5 system races. I'd like to see a small federation somewhere over in the Eastern side; maybe a couple overlords with one or more vassal civilizations; some other variations. 

But yeah, the important part is that all these races are there to provide the world-building material for the stories. If they don't help the stories move along, then there's not really a point to having them.

That being said, having a few extra-galactic areas (such as potentially Andromeda and the Pleides) would be good.

**Reply by AvianGeneticist** [reply]
I'm also of the opinion that map density should be greatly increased; obviously, the post above is just kind of a "what if" scenario type deal; I find it fun to think about these sorts of things and enjoy talking about them with others.

Though, I do have to question what the jump drive would be rebalanced like; should their ranged be reduced to reflect the rescale, or would it remain the same size, making it a vastly greater upgrade?

**Reply by warp-core** [reply]
> Though, I do have to question what the jump drive would be rebalanced like; should their ranged be reduced to reflect the rescale, or would it remain the same size, making it a vastly greater upgrade?

 I would expect the the background image to be larger; increasing the density across the whole galaxy would otherwise just make the map harder to navigate. The jump drive mechanics could then be maintained because, in terms of the distance the game considers (pixels in the map), the distances between most systems we have would (probably) be unchanged.

**Comment by leklachu** [comment]
If the density is decreased, would that mean just squashing human space... and in which case is consideration given to the real locations of stars (e.g. Polaris... I don't know if many/all systems are based on real stars?) Or adding extra systems in-between?

Changing human space game-mechanics-wise (e.g. adding many new systems in-between the ones we know) has big implications for the feel of the game.

Here's a radical idea... What if: sometime after the war, a new hyperdrive technology were discovered (I know, big and messy implications) that enabled reaching myriad systems in-between the ones we can currently reach? And... somehow breaking up the old links.... it'd need work. I don't think we have a 'lore' reason yet why we can reach some systems but not others (aside from 'these ones have links'): there's some scope to play around with there.

Amongst other questions is the question of habitable planets... most accessible systems have vaguely-habitable planets. For a very different feel, imagine Endless Sky where every hyperlink is now fifty jumps through fifty uninhabited systems. It would be fun for some of us...! And greatly change how galaxy-wide a sub-story can comfortably be.

---

As to "where do we put" the addition of alien species for now: I haven't followed other discussions, but it looks to me like there's a lot of galaxy left to go yet, if we expand out in all directions as far as human (and moreso wanderer/coalition) space does.

**Reply by warp-core** [reply]
> If the density is decreased, would that mean just squashing human space... and in which case is consideration given to the real locations of stars (e.g. Polaris... I don't know if many/all systems are based on real stars?) Or adding extra systems in-between?

Some stars have some correspondence to real life positions, others do not. Take Betelgeuse and Deneb, for example. Betelgeuse is ~550 lightyears away. Deneb, ~2600. And yet, Deneb appears far closer to Sol than Betelgeuse.

Though, the Orion constellation is fairly well represented: Betelgeuse, Rigel, Alnitak, Saiph, Mintaka, and Alnilam are all clustered relatively close together, with Bellatrix being much closer (but the real life star is much closer) but in approximately the same direction. Though, Alnitak is over twice as far from us as Betelgeuse and yet the distance to Sol for the two stars in game is relatively similar.

Zubeneschamali and Zubenelgenubi are both a part of the Libra constellation, and are fairly close to each other irl, but they are also fairly close to us, the closer of the two actually being closer to the Sun than it is to the other, and yet they are quite far away in game.

I haven't considered all the stars, but some thought certainly has gone into which ones should have which names. I suspect this is more based on how they appear from Earth and not necessarily where they actually are in the galaxy.

**Comment by ChuckStarchaser** [comment]
I've also arrived to the same conclusion as most people here.  That was EXACTLY my thinking, namely that the distances between linked systems is in the order of thousands of light years.  The question is how, exactly, to add density without breaking everything?  And perhaps I can contribute some ideas for discussion.
  Presently the map shows all systems connected by these mysterious space-lanes.  The fact that the space lanes are not the only way to travel is evident from the existence of jump-drives.
  Which means that the map that we have is the map **that the company that manufactures hyper-drives wants us to see,** as it is better for their business of hyper-drive sales.  Probably they are Syndicate, and they've lobbied the Republic to stay quiet on the issue.
  Once you have a jump-drive, you should be able to discover the true density of the galaxy... but not too fast.  You can only dial systems that are represented in your map/database.  You have a drive that can take you anywhere, but you don't have the data, yet, to tell it where to go.
  But you could meet some guy at a bat that after a few drinks tells you about a few systems that exist but are not in the standard map.
  Another thing that could be introduced slowly to the player is that the standard map is only a thin slice, and that there are systems above and below that plane;  specially towards the core of the galaxy, where the bulge is.
  Finally, as I suggested in my other ideas thread, we could make it so that it is possible to fly between stars without a special drive, if you are willing to wait very long, like 45 minutes of player time flying in a direction continuously, to reach another star.  This would give more of a sense of continuity to the map, would allow the placement of secret game assets away from the local system, and it would also allow the player to discover systems not represented in the official map.  Presumably, there would be pirate bases in some of the unofficial systems;  but not just pirates;  even secret bases belonging to corporations.  In the real world, for example,  a lot of the connoisseur illegal drugs, such as steroids, life extension antioxidants, antivirals that work (for the rich), etceteras, are actually made by the same companies that make our standard prescription drugs;  it's like a side business for them.  So, a modified (pirate) ship is probably made by the same company that makes the standard model, but just operates a factory somewhere off the official charts.
  There's a lot of opportunities for story-telling with a deeper map underlying a shallow, official map.
  And as a player you can discover this denser deep map slowly, by exploring, by hearsay, by missions that take you there, or by buying maps at pirate bases.

**Comment by vitalchip** [comment]
I noticed this while posting on another topic, lots of good thoughts and ideas bouncing around. Last time I played around with the code I actually ended up in the opposite direction.. making hyperspace travel much faster.

I was trying to solve a few dislikes I had and came to the conclusion that interstellar travel would have to take hours, not days and that having "a day" as the smallest unit of game time was no good.

I wanted to put an end to escorts travelling "time free"
I don't like the notion that all jumps take *a day*, for every ship, for every drive, for any link distance.

So now long links take longer to jump, fast ships travel the link faster and the clock ticks continuously throughout.

The problem comes in the balancing of time passing. Last figure I tried was 8 mins real time to a game day. This means that if you're in system mining, flying around or fighting etc then after 8 minutes it's tomorrow, it's more play time than it sounds but this I would like to be longer.

Using this time scale a shuttle can travel from Epsilon Leonis to Sol in 37 hours, a Flivver can do it in 19 hours.  These I would prefer to be longer.

The problem is these things are opposing, if I change the gametime/realtime ratio then as I make one longer the other becomes shorter.

The only way to satisfy both desires is to actually make jumping take more realtime and I've already increased that, at some point it becomes a lot of sitting around waiting for ships to complete the jump!

I like it generally, it brings a lot to the table, if you get a rush job then the natural choice becomes the smaller faster ship, a large slow ship just can't get there in time, unlike the standard where all ships complete a travel plan in the same game time. It means that ships can now actually fulfill a design purpose, the Bounder being a fast courier actually gets there faster (well, once it's top speed is corrected..)

It also means that calling in escorts from other systems takes game time, exactly the same time it would take you to jump there (assuming identical ship of course) plus the small, fast ships arrive much earlier than the large slow ships. 

You also can't cheat the time quite as much on a timed mission, if you jump ahead with a fast flagship you'll just have to wait at the far end for the slow ship carrying the mission cargo, all while the clock ticks. All ships have to make the deadline, not just the flagship.

**Reply by Zitchas** [reply]
That's an interesting idea. Kind of the opposite direction that I've favored, but I could see it being a pretty good way of doing it.

**Reply by bene-dictator** [reply]
Yes, I've thought about very similar things. For example: landing on a planet would fast-forward time by one hour. If you have to refuel, it'd take 1 hour per 500 fuel. Selling goods and buying new goods would take time; maybe, for ships like the container transport, it would be 1 hour per 50 tons, but the bulk freighter takes 1 hour per 15 tons. Visiting the spaceport, or buying outfits and ship might be another few hours. Being held up by pirates, accidentally buying too much, etc., that would feel more immersive. Maybe it can be frustrating (and now they would matter), but that's life in space! 

The only problem is scale, which you mention. I don't know about mechanics, but it won't be hard to explain something like "galactic time unit" where a minute in real life is 3 hours in-game (20 seconds to an hour). I would love to play a version of Endless Sky with this sort of time mechanic.

**Comment by ChuckStarchaser** [comment]
@vitalchip  Finer time subdivision sounds good, but it may take time for this idea to mature.  The one day per jump rule is something some players rely on to judge the difficulty of a time-limited mission.
  These missions are already too tight to be worth taking, at least from my perspective ...  I have evolved neural connections at the back of my retinas already, for fast detection of dates in mission descriptions, so I can reject them in smaller and smaller spans of milliseconds.  The way I make money is by taking dozens upon dozens of NON-time-limited travelers' missions, but to postpone their delivery until a good enough number of cargo missions force me to go to their destinations;  and then completing the cargo space with my own purchases.  Any timed mission at all is anathema to this optimization;  and so the only ones I take are the ones I take by cortical mistake or neural mis-click.  With a finer time-division and the lack of any way of knowing whether I have the means to complete any given mission in time, the undesirability of time-limited missions could reach cosmic heights.
  On the other hand, if you ask me, I'd simply get rid of time-limited missions altogether, as they are already a pain not unlike dog poop all over a side-walk.  All they do, at least for me, is add pain and frustration to the mission-selection process.
  But when I contemplate a future ES without timed missions I find an ES where the time dimension no longer matters, which is de-facto my situation, since I avoid them timed missions like the plague, so I couldn't care less what the date is.
  And so I would root for a deeper look into the time dimension before attempting to fix any problems with it:  Namely, a deep enquiry into the motivation for having a time dimension.  There should be more reasons for time than just timed missions.

**Reply by Zitchas** [reply]
a) this is a valid playstyle, and, I'll admit, one that I do from time to time, and especially in the early part of the game when I don't have much of the map revealed, and thus I have to feel my way across the galaxy, often running into dead ends or the like. That being said, in my own gameplay I find that about the time I have explored 50% of human space, I find myself shifting to prioritizing urgent or rush jobs, and finding more enjoyment in running specific urgent deliveries in a courier-style ship rather than doing bulk generic stuff.

b) That being said, figuring out ways to use time for stuff other than just timed missions is a valid point that we would do well to think on.

**Reply by ChuckStarchaser** [reply]
One game I used to play eons ago was  ...  Darn!  What was the name now?  ...  *Elite !*  That game was GOOD.  And interestingly it had dates,  and it had a running news drama you could read on the newspapers if you were so inclined.  And this was in spite of the fact that in space travel you had time-compression,  so your travels could take long, and yet they managed to have enough news drama for I don't know how long of game-time ...  could be a century, easily.  No obligation to read the newspapers, but sometimes the news were useful in planning your trading.  An earthquake in one planet, a famine in another, a war breaking out.  I may be off the mark, it was too many years ago so I remember almost nothing,  but the news were quite funny sometimes.  And it wasn't just daily newspapers;  there were weekly and monthly publications too, some by subscription, if memory serves me.

**Reply by Zitchas** [reply]
That's a great idea. And actually hearkens back to the original Escape Velocity trilogy, which had news items not terribly unlike ours - except instead of showing up in spaceports they had to be accessed by checking out the news button in the spaceport bar or showed up sometimes in merchant hails while in space. Otherwise, very similar. General news stories about random events "An industrial accident on Earth has driven up the price of medical supplies" or "A bumper crop on New Boston has bottomed out the market for food." Such things lasted about a week - just long enough that if you *really* want to, you could clear your cargo hold, load up on cargo, and head to the world to make a quick profit. (or head there to load up on cheap goods). It was a great way to encourage commodity running to different parts of human space, but didn't really make huge impacts on people running profitable routes.

They definitely made dealing with commodities more interesting, and made the world more alive. If we could tie those into our event system to do situational stuff, that could be even better, too. For instance, during the Civil War there could be new messages along the lines of "Rumors that the Navy is going to blockade Tarazed drive up food prices there." or directly tie into story events "A major battle in X has resulted in a high demand for equipment to use in repairs on the worlds of X, Y, Z."

We could do these latter things already, actually. But news stories about random worlds having short term increases/drops might take a bit more work.

**Reply by vitalchip** [reply]
I bought Elite in 1984, first game not shared with friends thanks to 'Lenslok'.  It had a lot of good stuff, things you had to think about and work at. Ships came poorly equipped and you had to buy things like auto refuellers and auto landers to make your life easier. Fuel cost money, quite a lot of money and longer jumps took more fuel so you had to factor the cost of getting there into your trades.

Trades were less certain too, there was no way to check prices in other markets, what you got was information about the planet and from there you'd have to imagine what goods might be high or low, that's if they were even available because unlike Endless Sky not everything was available every day, everywhere.

I remember having notes written down about good trade routes, lists of prices from previous visits etc. Having to refer to your own notes when thinking about the next trade run made it so immersive. While being able to see prices in advance for all the planets is probably 'realistic' because of, well, radio..  it does make things so predictable there's a lot less joy in it.

**Reply by Zitchas** [reply]
That seems like a trend... A lot of the older games, it seems, weren't afraid of making you refuel manually or actually do *all* the piloting or actually have to work to get ships into good optimization... I rather miss that.

**Comment by vitalchip** [comment]
Have to say I never found 'rush jobs' to be tricky, certainly they're not so easy if you want to take several of them as the combination of different directions/landings/jumps do tend to make some of them tight.

The general formula is that you get 2x jump days.

The real issue I have with it is that any ship at all can do a rush job, no need to have a fast ship. That's a missed trick for me. 

If I had a mission to take passengers to a destination within 24 hours then I could do it in an Arrow or a Flivver but a Shuttle or a Bulk Cargo ship would be too slow, it feeds into ships being made for a purpose and providing the job pays well enough gives the player a chance to have a use for different ships.

I'm not even suggesting this is a direction that should be investigated, it simply won't happen, this is just for my own 'version' The regular game hands everything to the player on a plate, they need to work for virtually nothing and there are exploits and loophoiles all over the place. I just enjoy closing them off and having to work for the rewards.

**Reply by Zitchas** [reply]
I agree, the fact that *any* ship can do any kind of job, including ultra-rush courier jobs, is a large part of the problem Endless Sky has with ship sizes. The inspiration series (EV/O/N) mitigated this by having small ships pass 1 day per jump, while medium and heavy ships passed multiple days. (I know this is old hat for you specifically, just mentioning it for others that might not be aware of it) The result was that only light ships could hope to do an urgent delivery, but a well built medium *might* be able to do a rush, and heavy ships could would be hard pressed to make the deadline on anything but the most forgiving of timed deliveries. This presented the player with a choice: They could make a lot of money running a tiny lightweight ship doing single jobs with very tight deadlines; or they could run a medium ship and take timed but not as urgent jobs with a bunch of regular jobs to pad their income; or they could do as ChuckStarchaser describes and get the biggest hauler they can get their hands on and load up on whole piles of non-time-sensitive jobs. It's a very simple mechanic, but one that made flying a small high-speed ship around the galaxy feel very rewarding and meaningful. 

It's actually part of the reason my pfp icon is the ship it is: The Starbridge, from EVN. It was basically one of the best light-category utility all-round ships one could buy: Capable of doing the fastest of urgent deliveries, while still being decent at combat and capacity. It was outclassed in every area except speed by most ships in the medium or heavy classes, but that was alright, because medium and heavies couldn't compete with it in terms of speed (both in-system manouvering or long-distance travel). Despite loving many of the tougher ships in the game (the Polaris Scarab in particular), it is the Starbridge that I kept coming back to as being what I considered emblematic of the most fun way to play the game. It constantly saddens me that, while we do  have a range of light ships in the game, there's little fun in flying them, and no reason to stick with them a second longer than it takes to afford the next biggest ship.

**Reply by ChuckStarchaser** [reply]
Indeed;  I've been touting the ship specialization angle of Masters of Orion II around here again and again, but that was specialization pertaining to combat.  It just hit me that specializations pertaining to transportation and courier ought to be equally if not more important.
  One thing to keep in mind, though, is that larger ships do not necessarily have to be slower in terms of speed;  it is more likely in terms of acceleration that they would hit a couple of walls.  Not only is acceleration more limited with larger structures, but even the same acceleration LOOKS lower on a larger vessel.  But in terms of absolute speed there is no logical reason to make a link between it and size.

**Reply by Zitchas** [reply]
That's true. Technically speaking in space there *is* no max speed, just a constantly increasing curve of how much force is required per unit of acceleration. Theoretically, every ship could reach any speed - but the larger the percentage of engine a ship has as a percentage of total mass, likely the faster it will be able to reach any given speed. A ship that is basically a giant engine with a guiddance system tacked onto it is probably going to reach the given target speed *long* before the ship for who the engine only takes up a quarter of its mass. Assuming equal engines, of course.

Practically speaking, though, we have maximum speeds. The AI doesn't seem to handle having infinite max speed very well; in part because most ships reach their max speed in a matter of seconds. This is one of the many reasons why I enjoy increasing the mass of ships across the board: Because this reduces their acceleration. If a fighter can reach its top speed in 5-10 seconds, and a freighter reaches its top speed in 30-60 seconds, the fighter is going to *feel* a lot faster, even if they both have a top speed of 2000.

**Reply by vitalchip** [reply]
I tried it both ways, first I had ships move into hyperspace based on their own acceleration and continue accelerating until they reached a count which was specific to that link and that ship (calculated from actual link length and ship acceleration) then they counted down again to emerge. That's a sort 'custom edition' of standard where the count and the acceleration are the same for all ships.

Although that worked and did have the desired effect I decide that I'd rather have it be based on acceleration and top speed and for hyperspace to 'properly' act as a distance compressor. So I have ship accelerate into hyperspace at a multiple of their normal acceleration and then stop accelerating when they hit a multiple of their top speed, cruise through hyperspace and when the appropriate link distance had been travelled they drop out and into the destination system.

This feels more like travelling through hyperspace and gives rise to overtaking or  being overtaken inside the jump by ships leaving before/after you with different characteristics.

For example if the flagship leaves at the same time as a slower accelerating ship it streaks away initially but if the other ship has a higher top speed and the distance to travel is enough it'll pass you "in hyperspace" 

A good practical use of this and one I found quite satisfying is that you can escort some cargo vessels in your faster flagship, jump together and you'll arrive in system and be able to deal with the waiting pirates while the cargo ships are still mkid jump and (hopefully) the area will be safe when they arrive.

**Reply by Zitchas** [reply]
That sounds almost like turning hyperspace into its own area or place to be and do stuff. I'm not sure how long I want to start at hyperspace animations if there isn't something to do while in it.

**Reply by vitalchip** [reply]
Yeh, it's a hard line to tread, everyone has their own threshold for what's going to feel like 'too long' I don't mind quite a lot of real time passing while playing, even if that is just transiting across a system or jumping but I'd imagine most people wouldn't feel the same.

It can really be any amount of real time though, hyperspace at the moment takes about 3 seconds of real time for any size link or ship. Making it take different amounts of real time can be balanced around an average but I was also looking at having game time pass while jumping so that comes with the other balancing issues from before.

**Reply by ChuckStarchaser** [reply]
@Zitchas The general SF problem with fictional unobtanium technologies and stuff is when the gobbledigok is brought into the lime-light and weaved into the foundations of a story, or worse yet, used to solve a hard problem that is central to a story.  This mortal sin was committed repeatedly by the story-writers for StarTrek, which is what made it such a cheap series as SF goes.  SF does not need to be hard science to be good;  it just has to not bring the gobbledigok to the foreground.  Example:  Azimov's Foundation Series makes use of impossible propulsion, relativity violations, and all kinds of impossible travel times;  but you don't even notice, because what matters to the reader is the story.  I think what vitalchip has done is to work out a formula for how hyperspace works INTERNALLY, which is more than just fine;  it's commendable  ... as long as the workings remain hidden from the player and are only understood intuitively after gaining experience.  It's just a way to make everything seem like there's rhyme, rhythm and reason to hyperspace, as opposed to being a capricious state.
  Something like that was done in the game Upon The Silent Sea, formerly Vegastrike.  In Vegastrike you were in the cockpit of the ship as it accelerated and ... well  ...  It's a long story whether or not there was hyperspace, or what exactly this warp engine did, but essentially it accelerated exponentially.  That is to say, it took the same time to accelerate from 10 to 20 as it did from 100 to 200 or as it did from 1000000 to 2000000.  But there were various factors affecting maximum velocity and acceleration, and the acceleration decreased as you neared maximum velocity in a way that felt right.  Thus, from the moment you hit that warp engine switch, you'd see the speed indicator increasing, adding decades after decades of speed, and you hardly noticed any movement until the decades started to get really high and then it was ridiculous how fast you moved.  It was not hard science, but it felt at least acceptable enough to forget about it and get on with game-play.  Any less complexity might have affected the perception negatively.  The math was fine-tuned to subjective perfection, IMO.
  Similarly, here, I was in no way suggesting that there be no maximum speed.  In the real world there will always be maximum speed because you can only accelerate for so long with limited fuel, even if you use relativistic exit velocity propulsion, such as VASIMR.  The ultimate question is what makes for the best game-play in the opinion of your target players.  And if your target players are like most people, they will want space flight that FEELS believable;  and it definitely cannot FEEL right if it is too mathematically simple.
  But I'm totally on board with NOT having animations that might bring the internal intricacies to the foreground.  If you do, then you invite skepticism and questions.  To have any kind of animations, the math needs an extra order of complexities to make the animation itself believable, and not too noticeably inconsistent with the actual internal math.
